haystack_experimental/dataclasses/chat_message.py

```diff
@@ -0,0 +1,136 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from dataclasses import asdict, dataclass, field
+from enum import Enum
+from typing import Any, Dict, List, Optional
+
+
+class ChatRole(str, Enum):
+    """Enumeration representing the roles within a chat."""
+
+    ASSISTANT = "assistant"
+    USER = "user"
+    SYSTEM = "system"
+    TOOL = "tool"
+
+
+@dataclass
+class ToolCall:
+    """
+    Represents a Tool call prepared by the model, usually wrapped in an assistant message.
+
+    :param id: The ID of the Tool call.
+    :param tool_name: The name of the Tool to call.
+    :param arguments: The arguments to call the Tool with.
+    """
+
+    id: str  # noqa: A003
+    tool_name: str
+    arguments: Dict[str, Any]
```

Comment:
`Dict[str, Any]` is very generic. I propose to iterate over it as the experiment proceeds.
Some providers return a JSON, while OpenAI returns a JSON dumped string...